### DISCUSSION TITLE
What modules can be used without Shizuku?

### DISCUSSION THREAD

**Discussion by e-t-l** [LSPosed/LSPatch#157] [Q&A]
I am running GrapheneOS, which unfortunately prevents Shizuku from running. Are there any modules that can still be used with LSpatch?

**Comment by wayner6** [comment]
You can use  every module without shizuku, because you can find the packaged apk（xxx.apk-lspatched) in the original directory of your apk file. Then you can install it directly.

**Reply by e-t-l** [reply]
Can you give any examples of modules that work fine without root or Shizuku? Because I have two modules installed and neither of them work.

**Reply by e-t-l** [reply]
Also, if "every module without Shizuku" worked, then there would be no reason for Shizuku to be integrated, so that's clearly not true. There's plenty of modules that modify the system in ways that require either Shizuku or root, because the whole point is it's beyond the capability of a normal app.

![Screenshot_20230111-120920_1](https://user-images.githubusercontent.com/40775958/211907771-fa362d37-da7a-40e2-ab75-f16b33152998.png)

LSpatch even states in the app that without Shizuku, "some functions are unavailable"

**Reply by wayner6** [reply]
You can ignore it,then you may find lspatched-apk in your phone.Install it directly.

Shizuku just provide a convenient way to install your app.

**Reply by Username10926663** [reply]
Shizuku is not needed I use force dark mode without Shizuku it worked well
